src/internal/connector/onedrive/restore.go

```diff
@@ -613,6 +613,14 @@ func getMetadata(metar io.ReadCloser) (Metadata, error) {
 // Augment restore path to add extra files(meta) needed for restore as
 // well as do any other ordering operations on the paths
 func AugmentRestorePaths(backupVersion int, paths []path.Path) ([]path.Path, error) {
+	// Nothing to do for versions where the directory metadata is stored in the
+	// directory it refers to as Corso will retrieve this data inline when
+	// restoring the collection for the directory. Path ordering does not matter
+	// for these versions.
+	if backupVersion == 0 || backupVersion >= version.OneDrive4DirIncludesPermissions {
```

Comment:
just double checking cause I'm juggling a few things at the moment: it's safe to exclude version 0 and not sort the paths correct?

Reply:
It should be safe to exclude 0, but we can't exclude >= 4 . We need to augment paths for newer items because of https://github.com/alcionai/corso/issues/2697 .

Reply:
good point, thanks for catching that and posting a PR of your own to fix things